packages/fetchai/connections/prometheus/connection.py

```diff
@@ -0,0 +1,314 @@
+                if metric_type is None:
+                    response_code = 404
+                    response_msg = (
+                        f"{message.type} is not a recognized prometheus metric."
+                    )
+                else:
+                    self.metrics[message.title] = metric_type(
+                        message.title, message.description
+                    )
+                    response_code = 200
+                    response_msg = (
+                        f"New {message.type} successfully added: {message.title}."
+                    )
+
+        elif message.performative == PrometheusMessage.Performative.UPDATE_METRIC:
+
+            metric = message.title
+            if metric not in self.metrics:
+                response_code = 404
+                response_msg = f"Metric {metric} not found."
+            else:
+                update_func = getattr(self.metrics[metric], message.callable, None)
+                if update_func is None:
+                    response_code = 400
+                    response_msg = f"Update function {message.callable} not found for metric {metric}."
+                else:
+                    # Update the metric
+                    update_func(message.value)
+                    response_code = 200
+                    response_msg = f"Metric {metric} successfully updated."
+
+        msg = dialogue.reply(
+            performative=PrometheusMessage.Performative.RESPONSE,
+            target_message=message,
+            code=response_code,
+            message=response_msg,
+        )
+        context = cast(EnvelopeContext, envelope.context)
+        envelope = Envelope(
+            to=msg.to,
+            sender=msg.sender,
+            protocol_id=msg.protocol_id,
+            message=msg,
+            context=context,
+        )
+        await self._send(envelope)
+
+    async def _send(self, envelope: Envelope) -> None:
+        """Send a message.
+
+        :param envelope: the envelope
+        :return: None
+        """
+        await self.queue.put(envelope)
+
+    async def disconnect(self) -> None:
+        """
+        Disconnect.
+
+        :return: None
+        """
+        if self._queue is not None:
+            await self._queue.put(None)
+            self._queue = None
+
+    async def get(self) -> Optional[Envelope]:
+        """Get incoming envelope."""
+        return await self.queue.get()
+
+
+class PrometheusConnection(Connection):
+    """Proxy to the functionality of prometheus."""
+
+    connection_id = PUBLIC_ID
+
+    def __init__(self, **kwargs):
+        """
+        Initialize a connection to a local prometheus server.
+
+        :param kwargs: the keyword arguments of the parent class.
+        """
+        super().__init__(**kwargs)
+
+        print(self.configuration)
+
+        self.port = cast(int, self.configuration.config.get("port", DEFAULT_PORT))
+        self.metrics = {}
```

Comment:
don't see why that needs instantiating here.

Reply:
Are you suggesting that `metrics` be initialized to `None` or that it only needs to be defined within the channel? If the latter, couldn't there be some scenario where the channel gets disconnected but you don't want to lose the metrics that have already been created?

Reply:
I think the latter scenario cannot happen given how the connections are used. If the channel crashes the connection does to.